frontend/src/useFlow.ts

```diff
@@ -247,25 +307,34 @@ export function useFlow() {
 
       // Save translated segments to database
       return {
-        translatedSegments: await buildAndSaveSegments(translated_paragraphs, translatedSourceId, properties, originalSegments),
+        translatedSegments: await buildAndSaveSegments(translated_paragraphs, translatedSourceId, properties, originalSegments, additional_sources_segments),
         translatedSourceId,
       };
     } finally {
       setLoadingCount(prev => prev - 1);
     }
   }, [setLoadingCount]);
 
-  const translateFile = useCallback(async (file: File, name: string, originalLanguage: string, translatedLanguage: string, stepByStep: boolean, dictionaryId: null|number, dictionaryTimestamp: null|string)
+  const translateFile = useCallback(async (file: File, name: string, originalLanguage: string, translatedLanguage: string, stepByStep: boolean, dictionaryId: null|number, dictionaryTimestamp: null|string, additionalSources: AdditionalSourceInfo[] = [])
     : Promise<{translatedSegments: Segment[], translatedSourceId: number}> => {
     console.log(dictionaryId, dictionaryTimestamp)
+    console.log('Additional sources:', additionalSources);
     setLoadingCount(prev => prev + 1);
     try {
-      const { originalSegments, translatedSourceId } = await initSourceFromFile(file, name, originalLanguage, translatedLanguage, dictionaryId, dictionaryTimestamp);
+      const { originalSegments, translatedSourceId, additionalSourcesText } = await initSourceFromFile(file, name, originalLanguage, translatedLanguage, dictionaryId, dictionaryTimestamp, additionalSources);
 
       // Step-by-step translation (first 10 paragraphs only)
       const originalSegmentsChunk = stepByStep ? originalSegments.slice(0, 10) : originalSegments;
 
-      return await translateSegments(originalSegmentsChunk, translatedSourceId, originalLanguage, translatedLanguage);
+      // Adjust additional sources text length to match combined originalSegmentsChunk length * multiplier
+      const combinedOriginalTextLength = originalSegmentsChunk.reduce((sum, segment) => sum + segment.text.length, 0);
+      const targetLength = Math.floor(combinedOriginalTextLength * ADDITIONAL_SOURCES_TEXT_LENGTH_MULTIPLIER);
+      
+      additionalSourcesText.forEach(result => {
+        result.text = result.text.substring(0, targetLength);
+      });
+
+      return await translateSegments(originalSegmentsChunk, translatedSourceId, originalLanguage, translatedLanguage, additionalSourcesText);
```

Comment:
You don't need `additionalSourcesText` as input here, they are written into the database.